chore: pin conda python deps

btraven00 opened the pull request:
We had already pinned python3.12 in master (mostly due to the pydantic issue #109), so porting that change back to dev.

Additionally, I'm providing a convenience script to setup a conda environment with micromamba.

imallona commented on scripts/create-micromamba-env.sh:
I think this is a bad idea as we discussed, we don't know what's on the user's $PATH. I think it would be better to recommend installation via miniconda or miniforge.

btraven00 replied:
on the other hand, let's be aware that the script was isomorphic to the documentation :smiling_imp:

imallona commented on scripts/create-micromamba-env.sh:
```diff
@@ -0,0 +1,11 @@
+# This script creates a micromamba environment called "omnibenchmark"
+# and installs the pinned system dependencies and an editable version of the
+# omnibenchmark python package.
+MAMBA=micromamba
```
micromamba is not friendly to neither mamba nor conda commands, I think this $MAMBA variable is confusing. We should perhaps assume conda is installed and just conda create the env; or suggest using miniconda or something more normal with a conda bundled in.

imallona replied:
still, `test-environment.yml` is necessary for ci/cd tests for fast env / conda management, but then has the caveat of having to install `conda` (or `mamba`) inside.

for an user-facing recipe I'd go for miniconda or miniforge.

imallona reviewed (changes requested):
See comments above

btraven00 commented:
I have reverted the instructions in the tutorial, and remove any reference to the script. For dev having a script is useful for me tho, since it saves some time. If we find it confusing to have it there I'm fine moving it somewhere else outside of the tree.

Since I was there, I removed a couple of things for the tutorial, like the mention to poetry, since it's not really needed anymore (in fact, python does install poetry as a build dependency if installing the folder, but there's no need to make it explicit, pip handles it underneath).

Re. changing the docs to use minimamba: maybe, once we're reasonably certain that it works. In any case, I'd just add a section as "recommended" way, and perhaps note that others work. We can do that in a new PR, but I'd make sure we switch the main tests to also use miniconda at the same time.

If any of  this is problematic, I'd rather split the PR into just the test-environment change and leave docs for a separate discussion.
